### ISSUE TITLE
[Question] c.Lang does not seem to be working for me. Could you please help?

### ISSUE THREAD

**Issue by qerraniket12345qwerty** [twintproject/twint#1053]
# Issue Template
Please use this template!

### Initial Check
> If the issue is a request please specify that it is a request in the title (Example: [REQUEST] more features). If this is a question regarding 'twint' please specify that it's a question in the title (Example: [QUESTION] What is x?). Please **only** submit issues related to 'twint'. Thanks.

>Make sure you've checked the following:

- [] Python version is 3.6;
- [] Updated Twint with `pip3 install --user --upgrade -e git+https://github.com/twintproject/twint.git@origin/master#egg=twint`;
- [] I have searched the issues and there are no duplicates of this issue/question/request.

### Command Ran
>Please provide the _exact_ command ran including the username/search/code so I may reproduce the issue.

c = twint.Config()
c.Lang = "en"
c.Popular_tweets = True
c.Limit = 5000
c.Since = "2018-11-30"
c.Until = "2020-11-30"
c.Search = 'GSK'
c.Store_csv = True
c.Output = "GSK.csv"
twint.run.Search(c)

### Description of Issue
>Please use **as much detail as possible.**

The language filter seemed to be working like a week back but doesn't seem to be working anymore.  The CSV file generated explicitly mentions other languages in the language column as well. The language column specifies other languages. Attached a screenshot of the same.
![Issue](https://user-images.githubusercontent.com/53196315/101174545-08fa8080-366a-11eb-808a-091197f293e4.png)

### Environment Details
>Using Windows, Linux? What OS version? Running this in Anaconda? Jupyter Notebook? Terminal?

Using Windows 10, Jupyter Notebook in Anaconda Environment.

**Comment by antoinecomp** [comment]
Neither for me, I think you need to do it in post production.

**Comment by ajctrl** [comment]
here is the fix!
https://github.com/twintproject/twint/pull/1025

**Comment by antoinecomp** [comment]
@ajctrl I just tried to do what they mention in the issue: replacing lines 102 and 103 of `url.py` with

```
    if config.Lang:
        q += f" lang:{config.Lang}"
```

but it didn't work for me. Indeed, when I do:

```
(venv) C:\Users\antoi\PycharmProjects\pythonProject>python
Python 3.6.7 (v3.6.7:6ec5cf24b7, Oct 20 2018, 13:35:33) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import twint
>>> c= twint.Config()
>>> c.Lang = "fr"
>>> c.Search = "burberry"
>>> twint.run.Search(c)
```

I obtain all kind of languages. 

```
1334904272744431616 2020-12-04 17:55:54 +0100 <blondyneczka> Check out what I just added to my closet on Poshmark: Burberry Dress.  https://t.co/3pwzu62BwU via @posh
markapp #shopmycloset
1334904261231202307 2020-12-04 17:55:51 +0100 <pausefun> Deux nouveaux parfums pour Burberry en 2016 -  https://t.co/iJc9qfHYX4  https://t.co/X4GFWj5p6t
1334904039310389249 2020-12-04 17:54:58 +0100 <nutnalinna> กลิ่นเปรี้ยวอมหวานแบบ Burberry Her คือไม่ไหวจริงๆ อยู่บนตัวเรานี่กลายเป็นกลิ่นลูกอมเลยอ่ะ แต่ถ้าเปรี้ยวหวา
นไปทางโลลิต้าแบบ YSL Mon Paris นี่คือชอบสุดๆ ชอบมาก เสียดายฉันทำขวดแบ่ง 10 มิลแตกไปและ อดใช้เบย
1334903976341409794 2020-12-04 17:54:43 +0100 <xdarling_Lola> Burberry men’s gift ideas is pretty lit  this year �
1334903808887988224 2020-12-04 17:54:04 +0100 <ASharpDressedM1> Check out what I just added to my closet on Poshmark: Burberry Tan Nova Check Plaid Button Down Shirt
 L.  https://t.co/0tXVXw3mIm via @poshmarkapp #shopmycloset
...
```

But maybe I'm missing something. I'm doing this in locally the version of twint I cloned and even my basic modifications doesn't seem to change anything.

**Comment by ajctrl** [comment]
well, it worked for me.
maybe you need to install a modified version like this: "pip install ."

**Comment by antoinecomp** [comment]
Yeah! You were right @ajctrl 

By the way, I'm trying to create a file [mongodb.py](https://github.com/antoinecomp/twint/blob/main/twint/storage/mongodb.py) equivalent to elasticsearch.py in order to be able to save the Tweets in a MongoDB database. However I am not able to add new config objects, despite adding them to config.py. Would you have any idea why?

**Comment by himanshudabas** [comment]
if you are making changes to the twint code locally, you need to uninstall and reinstall twint using the following command when you are in the cloned git directory of twint:
```
pip install .
```
also you can use the below command instead if you are gonna be making changes repeatedly, using the below command you won't need to uninstall and reinstall twint whenever you make any changes to the twint code.
```
pip install -e .
```

**Comment by antoinecomp** [comment]
@himanshudabas Many thanks and sorry, I think it's a pretty dumb question, I am noew to participating in OS projects.
I guess I still need to reimport twint whenever I modify its code?

**Comment by himanshudabas** [comment]
> I guess I still need to reimport twint whenever I modify its code?

Sorry I don't understand what you mean by _reimport twint_.
Do you mean you are using it in colab/jupyter notebook?

**Comment by antoinecomp** [comment]
@himanshudabas No in a terminal.
I meant to do
```
>>> exit()
: python
>>> import twint?
```
each time I do modifications to the source code.

**Comment by himanshudabas** [comment]
Yes, I suppose you do need to reimport it. I'm not 100% sure though, but it's logical to reimport it in this case.
I found a related [article here](https://support.enthought.com/hc/en-us/articles/204469240-Jupyter-IPython-After-editing-a-module-changes-are-not-effective-without-kernel-restart?mobile_site=true).
It's because even when you do _**import twint**_ again, without restarting your python terminal, python thinks _i already know this package, I don't need to read the code again_.
Article also talks about possible workarounds.

Though I'd suggest you to simply create a python script for testing twint. In that script you can import twint and write your entire script. If your script is let's say scrape.py, you can simply do `python scrape.py` and you don't need to write the entire script again, like you do in python terminal.
Also whenever you run this script, it automatically fetches the latest changes from the twint installation.

Do let me know if you need more help.

**Comment by westinbrevard** [comment]
I don't think the language feature is working either.

    On Saturday, December 5, 2020, 8:17:55 AM EST, Himanshu Dabas <notifications@github.com> wrote:  
 
 


Yes, I suppose you do need to reimport it. I'm not 100% sure though, but it's logical to reimport it in this case.
I found a related article here.
It's because even when you do import twint again, without restarting your python terminal, python thinks i already know this package, I don't need to read the code again.
Article also talks about possible workarounds.

Though I'd suggest you to simply create a python script for testing twint. In that script you can import twint and write your entire script. If your script is let's say scrape.py, you can simply do python scrape.py and you don't need to write the entire script again, like you do in python terminal.
Also whenever you run this script, it automatically fetches the latest changes from the twint installation.

Do let me know if you need more help.
